ProcessesApi/V1/Services/SoleToJointService.cs

```diff
@@ -102,13 +102,26 @@ private async Task OnProcessClosed(Stateless.StateMachine<string, string>.Transi
         {
             var processRequest = x.Parameters[0] as ProcessTrigger;
 
-            SoleToJointHelpers.ValidateFormData(processRequest.FormData, new List<string>() { SoleToJointFormDataKeys.HasNotifiedResident });
+            if (processRequest.FormData.ContainsKey(SoleToJointFormDataKeys.Reason))
+            {
+                SoleToJointHelpers.ValidateFormData(processRequest.FormData, new List<string>() { SoleToJointFormDataKeys.HasNotifiedResident, SoleToJointFormDataKeys.Reason });
+                var expectedFormDataKeys = new Dictionary<string, object>()
+                {
+                    { SoleToJointFormDataKeys.Reason, processRequest.FormData[SoleToJointFormDataKeys.Reason] }
+                };
+                _eventData = expectedFormDataKeys;
+            }
+            else
+            {
+                SoleToJointHelpers.ValidateFormData(processRequest.FormData, new List<string>() { SoleToJointFormDataKeys.HasNotifiedResident });
```

Comment:
You could move this above the if statement as FormData should always(?) contain `HasNotifiedResident`. Then you don't need to validate the formData inside the if statement because you've already checked if it contains `Reason`.